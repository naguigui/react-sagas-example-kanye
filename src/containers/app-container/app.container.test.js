import { AppContainerComponent } from "./app.container";
import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import { shallow } from "enzyme";

describe("AppContainer", () => {
  describe("rendering", () => {
    it("renders with a quote when isLoading is false and isError is false", () => {
      const quote = "I am a space cowboy.";
      const shallowRenderer = createRenderer();
      const tree = shallowRenderer.render(
        <AppContainerComponent
          isLoading={false}
          isError={false}
          quote={quote}
        />
      );
      expect(tree).toMatchSnapshot();
    });
    it("renders the loading state when isLoading is true and isError is false", () => {
      const shallowRenderer = createRenderer();
      const tree = shallowRenderer.render(
        <AppContainerComponent isLoading={true} isError={false} quote="" />
      );
      expect(tree).toMatchSnapshot();
    });
    it("renders the error state when isLoading is false and isError is true", () => {
      const shallowRenderer = createRenderer();
      const tree = shallowRenderer.render(
        <AppContainerComponent isLoading={false} isError={true} quote="" />
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe("getKanyeQuote()", () => {
    it("calls kanyeQuoteRequest function when invoked", () => {
      const kanyeQuoteRequestMock = jest.fn();
      const wrapper = shallow(
        <AppContainerComponent
          isLoading={false}
          isError={true}
          quote="Hello"
          kanyeQuoteRequest={kanyeQuoteRequestMock}
        />
      );
      wrapper.instance().getKanyeQuote();
      expect(kanyeQuoteRequestMock).toHaveBeenCalledTimes(1);
    });
  });
  describe("componentDidMount", () => {
    it("calls the instance method getKanyeQuote()", () => {
      const wrapper = shallow(
        <AppContainerComponent
          isLoading={false}
          isError={true}
          quote="Hello"
          kanyeQuoteRequest={() => {}}
        />
      );
      wrapper.instance().getKanyeQuote = jest.fn();
      wrapper.instance().componentDidMount();
      expect(wrapper.instance().getKanyeQuote).toHaveBeenCalledTimes(1);
    });
  });
});
