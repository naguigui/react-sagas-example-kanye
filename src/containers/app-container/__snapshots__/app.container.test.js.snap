// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer rendering renders the error state when isLoading is false and isError is true 1`] = `
<div
  className="app-container-wrapper"
>
  <span>
    Oh noes there was an error!
  </span>
</div>
`;

exports[`AppContainer rendering renders the loading state when isLoading is true and isError is false 1`] = `
<div
  className="app-container-wrapper"
>
  <LoadingSpinner />
</div>
`;

exports[`AppContainer rendering renders with a quote when isLoading is false and isError is false 1`] = `
<div
  className="app-container-wrapper"
>
  <div
    className="app-container-content"
  >
    <span
      className="app-container-quote"
    >
      I am a space cowboy.
    </span>
    <RippledComponent
      className=""
      disabled={false}
      initRipple={[Function]}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      unbounded={false}
    >
      Another one
    </RippledComponent>
  </div>
</div>
`;
